content/6_Advanced/Wavelet.mdx

```diff
@@ -32,9 +30,232 @@ Like a segment tree on values rather than indices.
 	</Resource>
 </Resources>
 
-### Solution - Range K-th Smallest
+Suppose you want to support the following queries:
+
+- Given a range $l$, $r$ count the number of occurrences of value $x$.
+- Given a range $l$, $r$ find the $k$ smallest element
+
+With wavelet tree, you can easily support those queries in $log(M)$ time.
+when M is the maximum value in the array.
+
+## Wavelet tree structure
+
+A wavelet tree is a binary tree where each node represents a range of values.
+The root node covers the entire range, and each subsequent level splits the
+range into two halves.
+
+We are going to maintain a segment tree over values instead of indices. Each segment will contain indices whose values lie within the segment's range. We'll save those indices in a vector. Notice that an index can be in at most $log(M)$ segments
```

Comment:
you can combine these two paragraphs since rurq is a prereq

also might be worth leading the reader to why we'd want to segtree over values instead of indices (why?), but it's your choice on how you want to explain it

Reply:
That's true, but it's a bit different. I want to emphasize that we are maintaining segments over values and explain what that means, so they won't confuse it with a standard segment tree.

as for your second comment I agree that could be useful information, I will try to add it.